provider/v2v_vmcheck_helper.py

```
@@ -443,3 +450,74 @@ def check_graphics(self, param):
+            logging.info('Skip Checking genid: %s' % reason)
+            return
+
+        supported_output = ['libvirt', 'local', 'qemu']
+        # Skip checking if any of them is not in supported list
+        if self.params.get('output_mode') not in supported_output:
+            reason = 'output_mode is not in %s' % supported_output
+            logging.info('Skip Checking genid: %s' % reason)
+            return
+
+        logging.info('Checking genid info in xml')
+        logging.debug('vmxml is:\n%s' % self.vmxml)
+        if has_genid == 'yes':
+            mount_point = utils_v2v.v2v_mount(self.vmx_nfs_src, 'vmx_nfs_src')
+            # For clean up
+            self.mount_records[len(self.mount_records)] = (
+                self.vmx_nfs_src, mount_point, None)
+
+            cmd = "cat {}/{name}/{name}.vmx".format(
+                mount_point, name=self.original_vm_name)
+            cmd_result = process.run(cmd, timeout=20, ignore_status=True)
+            cmd_result.stdout = results_stdout_52lts(cmd_result)
+            genid_pattern = r'vm.genid = "(-?\d+)"'
+            genidX_pattern = r'vm.genidX = "(-?\d+)"'
+
+            genid_list = [
+                re.search(
+                    i, cmd_result.stdout).group(1) if re.search(
+                    i, cmd_result.stdout) else None for i in [
+                    genid_pattern, genidX_pattern]]
+            if not all(genid_list):
+                logging.info(
+                    'vm.genid or vm.genidX is missing:%s' %
+                    genid_list)
+                # genid will not be in vmxml
+                if re.search(r'genid', self.vmxml):
+                    self.log_err('Unexpected genid in xml:\n%s' % self.vmxml)
+                return
+
+            genid_str = _compose_genid(*genid_list)
+            logging.debug('genid string is %s' % genid_str)
+
+            if not re.search(genid_str, self.vmxml):
+                self.log_err('Not find genid or genid is incorrect')
```

I assume it need fail ,rather than just log error information

---

V2V can record all failed checkpoints, and prints them out at last. the log_error function can implement that.